### ISSUE TITLE
ZDCChatViewController popped twice when dismissed

### ISSUE THREAD

**Issue by zipme** [zendesk/zendesk_sdk_chat_ios#82]
Hi,

When I push the chat vc onto my current navigation controller:
```swift
ZDCChat.startChatIn(self.navigationController, withConfig: nil)
```

and then click the back button on ZDCChatViewController, the view controller is popped twice.

I guess it's because when the vc is popped, `ZDCChat.endChat()` is called which triggers another pop operation. 

My current workaround is assign a `UIBarButtonItem` to chatViewController's leftBarButtonItem and call ZDCChat.endChat(), which is very ugly. 

Any idea why this happen?

Thanks!

**Comment by StevenDiviney** [comment]
@zipme Thanks for bringing this to our attention. You are correct, we explicitly dismiss the view controller in endChat. This appears to happen after the system dismissal. I have created an issue internally to track this and we will update you when it is fixed.

**Comment by oarrabi** [comment]
Hello @zipme,

We just released `ZDCChat` version 1.2.1.1 which has a fix to this issue.
Please check it out and let us know.

Thanks,
Omar...

**Comment by zipme** [comment]
@oarrabi It works perfectly! Thanks for the fix!

**Comment by cheah** [comment]
Seems like this is back in v1.3.0.2

**Comment by zipme** [comment]
Confirmed, it's back 😢

**Comment by baz8080** [comment]
Hi @cheah @zipme 

Sorry to hear that it is back. We implemented an alternative fix in 1.3.0.2 which is not working for you. Can I ask [how you are starting the chat](https://developer.zendesk.com/embeddables/docs/ios-chat-sdk/chat#starting-a-chat-in-a-navigation-controller) (modally / non-modally) please?

**Comment by zipme** [comment]
Hi @baz8080, I do it like this:
```
ZDCChat.startChatIn(self.navigationController, withConfig: nil)
```

as showed in the documentation.

**Comment by zipme** [comment]
Hi @baz8080 , any idea when this will be fixed?

**Comment by baz8080** [comment]
Hi @zipme 

We are picking this up in our next sprint. I'll update you when we have something more concrete. In ball-park terms it would be weeks, not months.

**Comment by zipme** [comment]
@baz8080 cool, thanks for the update!

**Comment by oarrabi** [comment]
@zipme, @cheah.

We just released Chat SDK version 1.3.1.1 which contains a fix to this issue. Please check it out and let us know if the issue persists.

Thanks :)

**Comment by cheah** [comment]
Seems to work now! Thank you!
